### PR TITLE
Strassen

### PR DESCRIPTION
**Pull request by AmandaStromdahl**
Added a complexity evaluation of strassen(), which belongs to the divide_and_conquer algorithms.

### CODE REVIEW

File: complex_functions/strassen.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
```
Comment (bastienfaivre) [inline complex_functions/strassen.py:1]: Same as your last PR, you can add the comment stating that this function is not ours

File: complex_functions/strassen.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import math
+
+def strassen(matrix1: list, matrix2: list) -> list:
+    """
```
Comment (bastienfaivre) [inline complex_functions/strassen.py]: For a better presentation of the file, you can remove these examples

File: report.md [hunk 3]
```diff
@@ -27,38 +27,59 @@ for each project, along with reason(s) why you changed to a different one.
...
-| Function7                                                           | Location7                        | CCN7        | CCN7man1                | CCN7man2                | NLOC7        |
-| Function8                                                           | Location8                        | CCN8        | CCN8man1                | CCN8man2                | NLOC8        |
+| Function                                      | Location                                             | CCN, Lizard | CCN, manual (Student 1) | CCN, manual (Student 2) | NLOC, Lizard |
+| --------------------------------------------- | ---------------------------------------------------- | ----------- | ----------------------- | ----------------------- | ------------ |
+| dj_oracle(case, num_qubits)                   | 9                                                    | 9           | -                       | 25                      |
+| strassen(matrix1, matrix2)                    | divide_and_conquer/strassen_matrix_multiplication.py | 12          | 10                      | -                       | 39           |
```
Comment (bastienfaivre) [inline report.md]: I found the same CCN ! You can add the same result in "manual student 2"

**Review by bastienfaivre — COMMENTED** [review]
Well done, please have a look on my few comments

**Comment by AmandaStromdahl** [comment]
Great suggestions, I've implemented all of them now.

File: report.md [hunk 4]
```diff
@@ -27,38 +27,59 @@ for each project, along with reason(s) why you changed to a different one.
...
-| Function6                                                           | Location6                        | CCN6        | CCN6man1                | CCN6man2                | NLOC6        |
-| Function7                                                           | Location7                        | CCN7        | CCN7man1                | CCN7man2                | NLOC7        |
-| Function8                                                           | Location8                        | CCN8        | CCN8man1                | CCN8man2                | NLOC8        |
+| Function                                      | Location                                             | CCN, Lizard | CCN, manual (Student 1) | CCN, manual (Student 2) | NLOC, Lizard |
+| --------------------------------------------- | ---------------------------------------------------- | ----------- | ----------------------- | ----------------------- | ------------ |
+| dj_oracle(case, num_qubits)                   | 9                                                    | 9           | -                       | 25                      |
```
Comment (bastienfaivre) [inline report.md]: The location is missing and the CCN are written in the wrong columns

**Review by bastienfaivre — COMMENTED** [review]
Just a last small fix to check

**Comment by AmandaStromdahl** [comment]
Good catch! It's fixed now.

**Comment by bastienfaivre** [comment]
LGTM

**Pull request merged**